ansible_base/activitystream/views.py

```diff
@@ -10,14 +10,37 @@
 from ansible_base.lib.utils.views.permissions import IsSuperuser
 
 
+def calculate_filter_backends():
+    """
+    Calculate the filter backends for the Activity Stream.
+
+    We want to use the ActivityStreamFilterBackend for the FieldLookupBackend, but we want to keep the
+    default filter backends (in order) for everything else.
+    """
+    default_filter_backends = AnsibleBaseDjangoAppApiView.filter_backends
+    filter_backends = []
+    added_activity_stream_backend = False
+    for backend in default_filter_backends:
+        if backend.__name__ == 'FieldLookupBackend':
+            filter_backends.append(ActivityStreamFilterBackend)
+            added_activity_stream_backend = True
+        else:
+            filter_backends.append(backend)
+
+    if not added_activity_stream_backend:
+        filter_backends.append(ActivityStreamFilterBackend)
```

Comment:
If I'm reading this right you are basically subbing `FieldLookupBackend` for `ActivityStreamFilterBackend` right?

Can we shorten this to just:
```python
if FieldLookupBackend in default_filter_backends:
    default_filter_backends.remove(FieldLookupBackend)
if ActivityStreamFilterBackend not in default_filter_backends:
    default_filter_backends.append(ActivityStreamFilterBackend)
```

Or something along those lines?

Reply:
I wanted to retain order, just in case. I think DRF will apply the filtering backends consecutively and narrow the queryset further each time, so the order could matter in theory I think in some cases.